cluster/store/db.go

```diff
@@ -112,8 +112,12 @@ func (db *localDB) UpdateClass(cmd *command.ApplyRequest, nodeID string, schemaO
 		meta.Class.VectorIndexConfig = u.VectorIndexConfig
 		meta.Class.InvertedIndexConfig = u.InvertedIndexConfig
 		meta.Class.VectorConfig = u.VectorConfig
+		meta.Class.ReplicationConfig = u.ReplicationConfig
 		meta.Class.MultiTenancyConfig = u.MultiTenancyConfig
 		meta.ClassVersion = cmd.Version
+		if req.State != nil {
+			meta.Sharding = *req.State
+		}
```

Comment:
**question:** is this intended ? to update sharding state with replication factor from here ?

Reply:
Yes, because replication factor changes the sharding state for the class